### ISSUE TITLE
npm install errors

### ISSUE THREAD

**Issue by giovannimanzoni** [mydesireiscoma/watosan.org#17]
Hello, see my log...

gio@pcgio:/media/nfs/www/vueTest/blog/watosan$ npm i

> bcrypt@3.0.0 install /media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Pre-built binaries not installable for bcrypt@3.0.0 and node@8.15.0 (node-v57 ABI, glibc) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error Remote end closed socket abruptly. 
node-pre-gyp WARN Pre-built binaries not installable for bcrypt@3.0.0 and node@8.15.0 (node-v57 ABI, glibc) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error bad download 
make: ingresso nella directory "/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/build"
  CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.o
make: ingresso nella directory "/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/build"
  CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.o
  CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt.o
sed: impossibile leggere ./Release/.deps/Release/obj.target/bcrypt_lib/src/blowfish.o.d.raw: File o directory non esistente
rm: impossibile rimuovere './Release/.deps/Release/obj.target/bcrypt_lib/src/blowfish.o.d.raw': File o directory non esistente
make: *** [bcrypt_lib.target.mk:103: Release/obj.target/bcrypt_lib/src/blowfish.o] Error 1
make: uscita dalla directory "/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/build"
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.19.0-2-amd64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding/bcrypt_lib.node" "--module_name=bcrypt_lib" "--module_path=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding" "--napi_version=3" "--node_abi_napi=napi"
gyp ERR! cwd /media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt
gyp ERR! node -v v8.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding/bcrypt_lib.node --module_name=bcrypt_lib --module_path=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding --napi_version=3 --node_abi_napi=napi' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:126:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:915:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
node-pre-gyp ERR! System Linux 4.19.0-2-amd64
node-pre-gyp ERR! command "/usr/bin/node" "/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt
node-pre-gyp ERR! node -v v8.15.0
node-pre-gyp ERR! node-pre-gyp -v v0.10.2
node-pre-gyp ERR! not ok 
Failed to execute '/usr/bin/node /usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding/bcrypt_lib.node --module_name=bcrypt_lib --module_path=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding --napi_version=3 --node_abi_napi=napi' (1)
../src/bcrypt.cc: In function ‘void encode_salt(char*, u_int8_t*, char, u_int16_t, u_int8_t)’:
../src/bcrypt.cc:121:24: warning: ‘snprintf’ output may be truncated before the last format character [-Wformat-truncation=]
  snprintf(salt + 4, 4, "%2.2u$", logr);
                        ^~~~~~~~
../src/bcrypt.cc:121:10: note: ‘snprintf’ output between 4 and 5 bytes into a destination of size 4
  snprintf(salt + 4, 4, "%2.2u$", logr);
  ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/bcrypt.cc: In function ‘void bcrypt(const char*, const char*, char*)’:
../src/bcrypt.cc:264:29: warning: ‘snprintf’ output may be truncated before the last format character [-Wformat-truncation=]
  snprintf(encrypted + i, 4, "%2.2u$", logr);
                             ^~~~~~~~
../src/bcrypt.cc:264:10: note: ‘snprintf’ output between 4 and 5 bytes into a destination of size 4
  snprintf(encrypted + i, 4, "%2.2u$", logr);
  ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: ***  Nessuna regola per generare l'obiettivo "../src/bcrypt_node.cc", necessario per "Release/obj.target/bcrypt_lib/src/bcrypt_node.o".  Arresto.
make: uscita dalla directory "/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/build"
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Linux 4.19.0-2-amd64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding/bcrypt_lib.node" "--module_name=bcrypt_lib" "--module_path=/media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt/lib/binding" "--napi_version=3" "--node_abi_napi=napi"
gyp ERR! cwd /media/nfs/www/vueTest/blog/watosan/node_modules/bcrypt
gyp ERR! node -v v8.15.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN watosan.org@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bcrypt@3.0.0 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the bcrypt@3.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/gio/.npm/_logs/2019-03-11T09_19_50_757Z-debug.log

**Comment by 0x00000001A** [comment]
Thanks for the issue
This project currently is not actively maintained, but I'll try to fix this issue today

Thank you

**Comment by 0x00000001A** [comment]
@giovannimanzoni 
As far as I understand, it's a bcrypt issue
I found some recommedations to upgrade bcrypt package to the latest versions, so, I did it.
In the same time, I upgraded other project dependencies to the latest versions (incl. nuxtjs) except apollo-upload, it will be updated later.
Check out, if it works for you

p.s. for some reason, I was able to install the dependencies without any errors on my PC, without upgrading `bcrypt`. Maybe the following link will be userful in your case:
https://github.com/kelektiv/node.bcrypt.js/issues/632#issuecomment-409801915

Thank you
